main.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# bubble sort
+def buble(arr)
+  n_times = arr.length - 1
+  n_times.times do |i|
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[i] > arr[i + 1]
+  end
+  print arr
+end
+
+# bubble_sort_by
+def bubble_sort_by(arr)
+  n_times = arr.length - 1
+  n_times.times do |i|
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if yield(arr[i], arr[i + 1]).positive?
+  end
+  puts arr
```

Comment:
- [ ] The automated tests don't detect an answer here because you are not returning any value here, you are only displaying it. Please try returning only the `arr`